### DISCUSSION TITLE
Use Numlock as hotkey launching and minimizing Qalculate. Paste calculated result in place where cursor is

### DISCUSSION THREAD

**Discussion by nplus-user** [Qalculate/qalculate-gtk#495] [Ideas]
Asking for new features:
1) Set Numlock as a hotkey to call Qalculate from tray onto screen and minimizing back to tray restoring previous state of Numlock - if it was "On" then after pressing it still "On". This feature needs Qalculate be capable to minimize in system tray, but can be used with minimizing in taskbar.
2) Qalculate automatically pastes result of calculation in the place where cursor is in a text or sheet editor. You type 2+2 and after pressing Enter you get result "4" in Qalculate window and in text/sheet editor where cursor's blinking.
Is it possible? Both for Linux and Windows.

**Comment by cousteaulecommandant** [comment]
This is actually asking for a combination of multiple new features/behaviors, which I suppose need to be addressed separately.  Let's analyze them separately:
1. Run Qalculate in a special mode where hitting Enter will "do something" with the result and immediately close the app (unlike the "regular" behavior which is to display the result and wait for further input).  This seems feasible and could be done through a command line flag.
2. Upon termination, go back to the previous focused window and paste the qalculated value.  I don't think this is easy to, and it will be highly OS-dependent - not sure how easy it is to make a command that "pastes something".  Perhaps something to do with input methods.  (It is almost definitely a thing because applications such as on-screen keyboards exist.)
    * A much easier solution would be to make Qalculate copy the result (which it can already do, when you press Ctrl-Alt-C) and then you paste it yourself with Ctrl-V, but that's not as quick as you described.
3. Set a keybinding to launch Qalculate.  That's very easy to do on Linux (I have mapped the "calculator" key on my keyboard (mapped to "XF86Calculator") to command qalculate-gtk), but on Windows not that much.  For example, from what I understood, in order for the "Calculator" button to launch one app or another, I need to go to "Default apps" and choose the app as the default "Calculator" app.  But for the app to appear in the list, it needs to have identified itself with the OS as being able to handle the `calculator:` URI protocol.
    * Alternatively, use some software that re-maps the Calculator (or some other) keybinding to a custom action; I think such software exists.
4. Make that key the NumLock key.  Personally I'd avoid this, since it would make you lose the NumLock functionality.  I just tried to do this in Linux (XFCE) and yes, it launches Qalculate, but it also disables/enables NumLock every time.  I'd need to disable NumLock for it to work.  Better to use a different key or keybinding (calculator, Windows-Q, ScrollLock...)

Overall I think it's doable (1 and 3 look feasible, not sure about 2 and 4), but it seems a bit of overkill for the alternative (open Qalculate, type and hit enter, Ctrl-Alt-C, Alt-tab back to your app, Ctrl-V).

**Reply by hanna-kn** [reply]
The number of steps can be reduced if you activate calculate as you type and add a keyboard shortcut which both copies the result and closes the application (calculate as you type would not be necessary if expression calculation is added to the available shortcut actions).

For the paste part it seems more reasonable to use some kind of macro utility, and/or use qalc in a script (especially if supported by the editor).
